File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/ServersBuyNowError.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServersBuyNowError._

case class ServersBuyNowError (
  /* Always false for error responses. */
  success: Option[Boolean],
/* Human-readable error summary. */
  text: Option[String],
/* List of specific validation error messages. */
  errors: Option[List[String]])

object ServersBuyNowError {
  import DateTimeCodecs._

  implicit val ServersBuyNowErrorCodecJson: CodecJson[ServersBuyNowError] = CodecJson.derive[ServersBuyNowError]
  implicit val ServersBuyNowErrorDecoder: EntityDecoder[ServersBuyNowError] = jsonOf[ServersBuyNowError]
  implicit val ServersBuyNowErrorEncoder: EntityEncoder[ServersBuyNowError] = jsonEncoderOf[ServersBuyNowError]
}

File: OPENAPI-REST-API/openapi-client/zapier/models/ServersBuyNowError.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}success`,
                label: `Always false for error responses. - [${labelPrefix}success]`,
                type: 'boolean',
            },
            {
                key: `${keyPrefix}text`,
                label: `Human-readable error summary. - [${labelPrefix}text]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}errors`,
                label: `List of specific validation error messages. - [${labelPrefix}errors]`,
                list: true,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'success': bundle.inputData?.[`${keyPrefix}success`],
            'text': bundle.inputData?.[`${keyPrefix}text`],
            'errors': bundle.inputData?.[`${keyPrefix}errors`],
        }
    },
}
